src/server/game/AI/CreatureAI.cpp

```diff
@@ -56,60 +56,74 @@ void CreatureAI::Talk(uint8 id, WorldObject const* whisperTarget /*= nullptr*/)
     sCreatureTextMgr->SendChat(me, id, whisperTarget);
 }
 
-void CreatureAI::DoZoneInCombat(Creature* creature /*= nullptr*/, float maxRangeToNearestTarget /* = 50.0f*/)
+inline bool IsValidCombatTarget(Creature* source, Player* target)
 {
```

Comment:
Does it do anything significantly different from CanCreatureAttack() ? (with setting skipDistCheck to false)
![image](https://user-images.githubusercontent.com/2592673/140397378-06aa649f-c9a1-43c7-8de6-af4d4cbe17b8.png)

Reply:
Yes - it's different. `CanCreatureAttack` checks more things that are not needed here.